packages/react-testing/README.md

```diff
@@ -0,0 +1,267 @@
+# `@shopify/react-testing`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-testing.svg)](https://badge.fury.io/js/%40shopify%2Freact-testing.svg)
+
+A library for testing React components according to our conventions.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-testing
+```
+
+## Usage
+
+This library is broadly similar to [Enzyme](https://github.com/airbnb/enzyme), but with a focus on type safety and testing based on a component’s external API. In order to keep the API small and easy-to-use, it will generally track to only the latest minor release of React.
+
+### <a name="mount"></a> `mount(element: React.ReactElement<any>)`
+
+Mounts a component to the DOM and returns a [`Root`](#root) instance. Note that in order for this to work, you must have a simulated browser environment, such as the `jsdom` environment that Jest uses.
+
+Unlike Enzyme, there are no options to `render` or `shallow` render a component. These patterns do not work well under many circumstances, so there is no API in this library to use them.
+
+### <a name="destroyAll"></a> `destroyAll()`
+
+All mounted components are tracked in-memory. `destroyAll()` forcibly unmounts all mounted components and removes the DOM node used to house them. You should run this after each test that mounts a component (this is often done in a global `afterEach` hook).
+
+### <a name="root"></a> `Root<Props>`
+
+A `Root` object represents a mounted React tree. Most of the properties and methods it exposes are simply forwarded to the [`Element`](#element) instance representing the top-level component you rendered:
+
+- [#children](#children)
+- [#descendants](#descendants)
+- [#props](#props)
+- [#isDOM](#isDOM)
+- [#instance](#instance)
+- [#domNode](#domNode)
+- [#domNodes](#domNodes)
+- [#html()](#html)
+- [#text()](#text)
+- [#is()](#is)
+- [#prop()](#prop)
+- [#find()](#find)
```

Comment:
I mentioned below I'd prefer `find`, `filter` to `find`, `findAll`, but that didn't cover the difference between the regular and `where` prefixed versions.

I think here's roughly how I'd like to see this look

```
find(p: predicate<Element>): Element
filter(p: predicate<Element>): Element[]
querySelector(selector: string | ReactComponentType): Element
querySelectorAll(selector: string | ReactComponentType): Element[]
```

This way instead of basing our APIs on Enzyme they hew close to the standard JS APIs. It means a slightly more annoying time changing over to this library, but I think it is more grokkable to total newcomers and helps give our testing library it's own identity as a sensible solution.

Reply:
If `querySelector` is too verbose I would also accept 
- `query` / `queryAll`
- the library exporting some helper predicates `find(matchSelector('query'))`, `filter(matchComponent(MyComponent))`
- `find` and `filter` also accepting query selectors as an override
- `$` (JK)

Reply:
My rule of thumb here was to make the common thing the easiest thing, which is why I prefer having `.shortName(Component)` be available since we do this by far the most often. I am not totally opposed to query/ querySelector but it does feel odd because we do not allow you to pass in arbitrary DOM-ey selectors, even though you would probably expect that given the alignment of naming.

Reply:
Ah, I didn't catch that difference, that is a good reason not to use `querySelector` as a name, but also probably a good reason not to use the same name as enzyme's ([wrapper#find](https://airbnb.io/enzyme/docs/api/ReactWrapper/find.html) does allow dom style selectors).

I think the best option then is `find` and `filter` accepting `predicate | ComponentType`